### DISCUSSION TITLE
Presidio V2 is here!

### DISCUSSION THREAD

**Discussion by navalev** [microsoft/presidio#584] [General]
In the last couple of weeks we were busy working on the new version of Presidio.
The new version focuses on the Analyzer and Anonymizer, converting them to a python based pip installable package. We repalced gRPC with HTTP and allowed a more customizable APIs and easier debugging. We improved the documentation, added samples and build flows.
The Image Redactor (formerly presidio-image-anonymizer) is in early beta and is Python based and pip installable and other services will be migrated over time with the help of this community.

The legacy code base (aka V1) will continue to be available under the V1 branch but will no longer be officially supported.

We are certain this change will make Presidio much more accessible, maintainable and customizable and we look forward to collaborating with this great community!

**Comment by olivierpaling** [comment]
@navalev  Great work! At a first glance, the current API spec for v2 does not provide a way to add pattern-based recognizers 'on the fly' which was available in the v1 for blacklist + regex, is that correct? If so are there any plans to add this functionality in the near future or was this removed with the external recognizer store in v2. Thank you!

**Reply by omri374** [reply]
Hi @olivierpaling,
Thanks for reaching out!

You are correct that V2 does not have this capability. 
The reason is the types of deployments we would wish to support, which extend further than Kubernetes. 

The V1 recognizer store was using a persistent storage leveraging redis on Kubernetes. In the scenario where we deploy multiple analyzers, we would like to have new recognizers available for all  analyzers so all instances would interact with this store. 

On other types of deployment, this is getting tricky and generalization is challenging. The simple solution of exposing an API for the analyzer to do this would work for a single instance deployment, but at scale this would not work as you would have to somehow let all the analyzers know about the new recognizer and not just the one which answered the request for adding a recognizer. 

Having said that, we might re-introduce the recognizer store component in a future release, tailored for specific deployments such as Kubernetes.

Hope this helps clarifying the decision!

**Reply by omri374** [reply]
ad-hoc recognizers introduced in #677
